src/Microsoft.Data.Analysis/PrimitiveDataFrameColumnArithmetic.cs

```diff
@@ -6763,8 +6663,201 @@ public void ElementwiseLessThan(PrimitiveColumnContainer<ushort> column, ushort
+        {
+            throw new NotSupportedException();
+        }
+        public void And(PrimitiveColumnContainer<DateTime> column, DateTime scalar)
+        {
+            throw new NotSupportedException();
+        }
+        public void And(DateTime scalar, PrimitiveColumnContainer<DateTime> column)
+        {
+            throw new NotSupportedException();
+        }
+        public void Or(PrimitiveColumnContainer<DateTime> left, PrimitiveColumnContainer<DateTime> right)
+        {
+            throw new NotSupportedException();
+        }
+        public void Or(PrimitiveColumnContainer<DateTime> column, DateTime scalar)
+        {
+            throw new NotSupportedException();
+        }
+        public void Or(DateTime scalar, PrimitiveColumnContainer<DateTime> column)
+        {
+            throw new NotSupportedException();
+        }
+        public void Xor(PrimitiveColumnContainer<DateTime> left, PrimitiveColumnContainer<DateTime> right)
+        {
+            throw new NotSupportedException();
+        }
+        public void Xor(PrimitiveColumnContainer<DateTime> column, DateTime scalar)
+        {
+            throw new NotSupportedException();
+        }
+        public void Xor(DateTime scalar, PrimitiveColumnContainer<DateTime> column)
+        {
+            throw new NotSupportedException();
+        }
+        public void LeftShift(PrimitiveColumnContainer<DateTime> column, int value)
+        {
+            throw new NotSupportedException();
+        }
+        public void RightShift(PrimitiveColumnContainer<DateTime> column, int value)
+        {
+            throw new NotSupportedException();
+        }
+        public void ElementwiseEquals(PrimitiveColumnContainer<DateTime> left, PrimitiveColumnContainer<DateTime> right, PrimitiveColumnContainer<bool> ret)
+        {
+            for (int b = 0; b < left.Buffers.Count; b++)
+            {
+                var buffer = left.Buffers[b];
+                var mutableBuffer = DataFrameBuffer<DateTime>.GetMutableBuffer(buffer);
+                left.Buffers[b] = mutableBuffer;
+                var span = mutableBuffer.Span;
+                var otherSpan = right.Buffers[b].ReadOnlySpan;
+                for (int i = 0; i < span.Length; i++)
+                {
+                    ret[i] = (span[i] == otherSpan[i]);
+                }
+            }
+        }
+        public void ElementwiseEquals(PrimitiveColumnContainer<DateTime> column, DateTime scalar, PrimitiveColumnContainer<bool> ret)
+        {
+            for (int b = 0; b < column.Buffers.Count; b++)
+            {
+                var buffer = column.Buffers[b];
+                var mutableBuffer = DataFrameBuffer<DateTime>.GetMutableBuffer(buffer);
+                column.Buffers[b] = mutableBuffer;
+                var span = mutableBuffer.Span;
+                for (int i = 0; i < span.Length; i++)
+                {
+                    ret[i] = (span[i] == scalar);
```

Comment:
>span[i] == scalar [](http://example.com/codeflow?start=30&length=17)

How much we care comparing 2 DateTime objects which containing same number of ticks but created with different time zone? the remark [https://docs.microsoft.com/en-us/dotnet/api/system.datetime.op_equality?view=net-6.0#remarks](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.op_equality?view=net-6.0#remarks) explain more about that.

Reply:
This comment apply for other equality methods in this file too.

Reply:
Good question here too. I don't think this is a concern until we hear customer feedback that they need better equality support.

Reply:
Do we understand the user scenarios? How will the data get supplied? Changing that in the future can be a breaking change. Will be good if we understand the expectation and code for that.